lib/matplotlib/backends/backend_ps.py

```diff
@@ -1354,12 +1352,10 @@ def write(self, *kl, **kwargs):
                              rotated=psfrag_rotated)
 
             if is_writable_file_like(outfile):
-                if file_requires_unicode(outfile):
-                    with io.open(tmpfile, 'rb') as fh:
-                        outfile.write(fh.read().decode('latin-1'))
-                else:
-                    with io.open(tmpfile, 'rb') as fh:
-                        outfile.write(fh.read())
+                with (io.open(tmpfile, 'r', encoding='latin-1')
+                        if file_requires_unicode(outfile)
+                        else io.open(tmpfile, 'rb')) as fh:
```

Comment:
At the risk of bikeshedding... I find this 3-line construct a bit of a trip hazard when reading.  How about this:
```python
if file_requires_unicode(outfile):
    kwd = dict(mode='r', encoding='latin-1')
else:
    kwd = dict(mode='rb')
with io.open(tmpfile, **kwd) as fh:
```
Or if you want to reduce the line count:
```python
kwd = (dict(mode='r', encoding='latin-1') if file_requires_unicode(outfile)
            else dict(mode='rb'))
with io.open(tmpfile, **kwd) as fh:
```
The latter option is no longer than your original, and to me, much more readable. I haven't checked, but I hope it doesn't fall afoul of the line length limit.

Reply:
Yeah, I totally agree w/ this but thought maybe I was being an anti-fancy-python-tricks stick-in-the-mud

Reply:
OK

Reply:
Was this actually changed? It's not marked outdated.

Reply:
Ah, sorry.  There were two instances of it, only changed one of them.